root/etc/cont-init.d/45-plex-hw-transcode

```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/with-contenv bash
+
+# Check for the existence of the Intel video device
+if [ -e /dev/dri ]; then
+	VIDEO_GID=$(stat -c '%g' /dev/dri/* | head -n 1)
+else
+	exit 0
+fi
+
+# Check if this GID matches the current plex user
+PLEX_GID=$(getent group plex | awk -F: '{print $3}')
+if [ "${PLEX_GID}" == "${VIDEO_GID}" ]; then
+	exit 0
+fi
+
+# Check if the GID is taken and swap to 65533
```

Comment:
Before the rest of my comments, I have to ask: Why not just simply add the plex user to the GID of the /dev/dri/* device instead of creating groups and/or moving them around?  I cannot find documentation on whether `usermod -G` accepts gids but `-g` does so it likely does as well.  I did see that `-g` requires the group to exist so `-G` may as well.

Was 65533 arbitrarily chosen?  Either way the comment should describe why this value.

I had to look at the following code several times to understand it so I'll ask for more commenting.

Reply:
@gbooker the GID of the device gets reset to root on reboot of the host, so this is a more persistent way of resolving access to the device on container first deploy/restart. So the script fetches the current GID of the device and adds that into the `video` group, and then adds the `plex` user into the same group membership.

I'll add more comments to the functions as I understand them in the script, which I've ported from the original Linuxserver.io container that uses `abc` as the base user. I'll push another commit here for you to hopefully approve 👍

Reply:
Should be resolved by d3ba127